LLM/embedding/deploy-embedding-model-tei.md

```diff
@@ -0,0 +1,268 @@
+
+* Go to your tenancy Container Registry
+* Click on the Create repository button
+* Select Private under Access types
+* Set a name for Repository name. We are using "text-embeddings-inference" in the example.
+* Click on Create button
+
+You may need to docker login to the Oracle Cloud Container Registry (OCIR) first, if you haven't done so before in 
+order to push the image. To login, you have to use your API Auth Token that can be created under your 
+Oracle Cloud Account->Auth Token. You need to login only once. Replace with the OCI region you are using.
+
+```
+docker login -u '<tenant-namespace>/<username>' <region>.ocir.io
+```
+
+If your tenancy is federated with Oracle Identity Cloud Service, use the format /oracleidentitycloudservice/. 
+You can then push the container image to the OCI Registry.
+
+```
+docker tag ghcr.io/huggingface/text-embeddings-inference:1.5.0 -t <region>.ocir.io/<tenancy>/text-embeddings-inference:1.5.0
+docker push <region>.ocir.io/<tenancy>/text-embeddings-inference:1.5.0
+```
+
+# Prepare the model artifacts
+
+To prepare model artifacts for deployment:
+
+* Download the model files from Hugging Face to local directory using a valid Hugging Face token (only needed for gated models). If you don't have a Hugging Face Token, refer [this](https://huggingface.co/docs/hub/en/security-tokens) to generate one.
+* Upload the model folder to a [versioned bucket](https://docs.oracle.com/en-us/iaas/Content/Object/Tasks/usingversioning.htm) in Oracle Object Storage. If you don’t have an Object Storage bucket, create one using the OCI SDK or the Console. Make a note of the namespace, compartment, and bucketname. Configure the policies to allow the Data Science service to read and write the model artifact to the Object Storage bucket in your tenancy. An administrator must configure the policies in IAM in the Console.
+* Create model catalog entry for the model using the Object storage path.
+
+## Downloading model from Hugging Face Hub
+
+You can to [HuggingFace documentation](https://github.com/huggingface/huggingface_hub) for details. Here, we'll deploy one of the most popular embedding models from the hub that is supported by TEI.
+
+```
+# Login to huggingface
+huggingface-cli login --token "<your-huggingface-token>"
+ 
+# download the model to local folder. Here, BAAI/bge-base-en-v1.5 can be replaced by other models available on Hugging Face Hub.
+huggingface-cli download BAAI/bge-base-en-v1.5 --local-dir BAAI/bge-base-en-v1.5
+```
+
+## Upload Model to OCI Object Storage
+
+Once models are downloaded, use the terminal to upload the artifacts to object storage. Make sure that the bucket is versioned as mentioned above.
+
+```
+oci os object bulk-upload -bn <bucket> -ns <namespace> --auth resource_principal --prefix BAAI/bge-base-en-v1.5/ --src-dir BAAI/bge-base-en-v1.5/ --no-overwrite
+```
+
+# Create Model by reference using ADS
```

Comment:
I think we need to point users out from where they can do this? From local machines or NB session?

Reply:
added the details to use local (for docker), terminal for commands and notebook for code.